mm2src/coins/lightning.rs

```diff
@@ -206,14 +165,70 @@ impl LightningCoin {
             .pay_pubkey(destination, payment_preimage, amount_msat, final_cltv_expiry_delta)
             .map_to_mm(|e| SendPaymentError::PaymentError(format!("{:?}", e)))?;
         let payment_hash = PaymentHash(Sha256::hash(&payment_preimage.0).into_inner());
+        let payment_type = PaymentType::OutboundPayment {
+            destination: Some(destination),
+        };
 
-        Ok((payment_hash, PaymentInfo {
+        Ok(PaymentInfo {
+            payment_hash,
+            payment_type,
+            description: None,
             preimage: Some(payment_preimage),
             secret: None,
-            status: HTLCStatus::Pending,
             amt_msat: Some(amount_msat),
             fee_paid_msat: None,
-        }))
+            status: HTLCStatus::Pending,
+            created_at: now_ms() / 1000,
+            last_updated: now_ms() / 1000,
+        })
+    }
+
+    async fn get_open_channels_by_filter(
+        &self,
+        filter: Option<OpenChannelsFilter>,
+        paging: PagingOptionsEnum<u64>,
+        limit: usize,
+    ) -> ListChannelsResult<GetOpenChannelsResult> {
+        let mut total_open_channels: Vec<ChannelDetailsForRPC> = self
+            .channel_manager
+            .list_channels()
+            .into_iter()
+            .map(From::from)
+            .collect();
+
+        total_open_channels.sort_by(|a, b| a.rpc_channel_id.cmp(&b.rpc_channel_id));
+
+        let open_channels_filtered = if let Some(ref f) = filter {
+            total_open_channels
+                .into_iter()
+                .filter(|chan| apply_open_channel_filter(chan, f))
+                .collect()
+        } else {
+            total_open_channels
+        };
+
+        let offset = match paging {
+            PagingOptionsEnum::PageNumber(page) => (page.get() - 1) * limit,
+            PagingOptionsEnum::FromId(rpc_id) => open_channels_filtered
+                .iter()
+                .position(|x| x.rpc_channel_id == rpc_id)
+                .map(|pos| pos + 1)
```

Comment:
When querying from SQL using `FromId` we use the `OFFSET` clause with the row number of the `Id` as the offset, this is not inclusive as the `OFFSET` clause means skipping this number of rows. This is what I followed here to be the same as `list_closed_channels_by_filter` and other SQL querying methods https://github.com/KomodoPlatform/atomicDEX-API/blob/6ccb32feb9df40fa5a465adeba310d9723db0699/mm2src/coins/lightning.rs#L1054.

> Probably, your approach is better, so it's worth to choose how to process FromId in a uniform way.

I guess `FromId` is supposed to be used this way. The user would have got the info for the id used in `FromId` in the previous page/query and he would use this id (which is the last id returned in the page) to query the next page, so if it's inclusive there would be duplication in the results.

One last question. Are there any other cases other than `my_recent_swaps` and `account_balance` that need to be changed? I can work on them on this PR I guess but will this change break anything in the GUIs since these methods are probably already implemented?

Reply:
I've checked `my_recent_swaps` only because it's based on `offset_by_uuid` and it works inclusively.
`offset_by_uuid` is also used by `orders_history_by_filter` RPC.
>  Are there any other cases other than my_recent_swaps and account_balance that need to be changed?

Reply:
`FromId` should not be inclusive. I've just rechecked `my_recent_swaps` - it doesn't include `from_uuid` swap in the response.

Reply:
It seems to be my mistake. Can't reproduce it again, but... We found two issues :D
1) `my_recent_swaps` works incorrectly in WASM
2) `account_balance` need to be fixed as well

Reply:
Since these issues are not related to the PR, this comment can be resolved :)